src/Native/Runtime/amd64/InteropThunksHelpers.S

```diff
@@ -4,19 +4,43 @@
 
 .intel_syntax noprefix
 #include <unixasmmacros.inc>
-
-//TODO: Implement these stub method for non-windows
+#include <AsmOffsets.inc>         // generated by the build from AsmOffsets.cpp
+#define POINTER_SIZE 8
 
 LEAF_ENTRY RhpCommonStub, _TEXT
-    int 3
+
+    PUSH_ARGUMENT_REGISTERS
+
+    // +8 for alignment
+    alloc_stack    (SIZEOF_MAX_FP_ARG_SPILL + 8)
+    SAVE_FLOAT_ARGUMENT_REGISTERS 0 
+
+    INLINE_GET_TLS_VAR  tls_thunkData
+
+    mov    r11, [r10]
```

Comment:
@shrah Noticed one more problem: r10 and r11 are scratch registers so the call inside `INLINE_GET_TLS_VAR` can corrupt them. Could you please fix that?